ChebsNecromancy/Minions/NeckroGathererMinion.cs

```diff
@@ -195,36 +195,18 @@ private bool ReturnHome()
             return container.GetInventory().GetEmptySlots() < 1;
         }
 
-        private Container GetNearestDropOffPoint()
-        {
-            // find and return drop off point (some container with room)
-
-            // doesnt work, dunno why
-            //List<Piece> nearbyPieces = new List<Piece>();
-            //Piece.GetAllPiecesInRadius(transform.position, dropoffPointRadius.Value, nearbyPieces);
-            //
-            //if (nearbyPieces.Count < 1) return false;
-            Collider[] nearbyPieces = Physics.OverlapSphere(transform.position + Vector3.up, DropoffPointRadius.Value, pieceMask);
-            if (nearbyPieces.Length < 1) return null;
-
-            // order piece from closest to furthest, then take closest container
-            Collider closest = nearbyPieces
-                .OrderBy(piece => Vector3.Distance(transform.position, piece.transform.position))
-                .FirstOrDefault(piece =>
-                {
-                    var pieceContainer = piece.GetComponentInParent<Container>();
-                    return pieceContainer != null && pieceContainer.GetInventory().GetEmptySlots() > 0;
-                });
-            if (closest != null)
-            {
-                Container closestContainer = closest.GetComponentInParent<Container>();
-                if (closestContainer != null)
-                {
-                    // move toward that piece
-                    _monsterAI.SetFollowTarget(closest.gameObject);
-                    return closestContainer;
-                }
+        private Container GetNearestDropOffPoint() {
+            // Container closestContainer;
+            Container closestContainer = FindClosest<Container>(DropoffPointRadius.Value, pieceMask,
+                c => c.GetInventory().GetEmptySlots() > 0);
+            if (closestContainer == null) return null;
+            if (TryGetComponent(out MonsterAI monsterAI)) {
```

Comment:
I get and store a reference to `MonsterAI` in `Awake` called `_monsterAI`. Rather than retrieving it again each time, you can use that. This should slightly improve performance.

Feel free to disagree though and if you do, plz tell me why. I am always eager to learn something new.

Reply:
Makes sense! 😁

Reply:
Resolved in e9a0cf79d88357f9ba778477d65c0ef88ec9eb5f.

Reply:
Much nicer code, too, thank you!